upsetplot/data.py

```diff
@@ -16,3 +19,51 @@ def generate_data(seed=0, n_samples=10000, n_sets=3, aggregated=False):
     if aggregated:
         return df.value.groupby(level=list(range(n_sets))).count()
     return df.value
+
+
+def from_sets(data):
+    """
+    Data loader for a dict of sets
+
+    :param Dict[str, set[str]] data:
+    :return: Multi-Index Series of intersection counts
+    :rtype: pd.Series
+    """
+    # Convert dict into OrderedDict to preserve Key/Value order
+    data = OrderedDict(data)
+
+    # Construct Index
+    tf_array = [[True, False]] * len(data)
+    index = pd.MultiIndex.from_product(tf_array, names=data.keys())
+
+    # Curate values from each intersection group
+    values = []
+    for i in index:
+        values.append(_intersection_counts(data.values(), i))
```

Comment:
This operates in `O((2 ** n_sets) * n_items)` time. We can do it in `O(n_items)` time by just calculating the set memberships for each item once, then aggregating the count to the appropriate row.